File: Data Structures & Algorithms/validate-parentheses/submission-1.js
class Solution {
    /**
     * @param {string} s
     * @return {boolean}
     */
    isValid(s) {
        let n=s.length,stack=[];
        for(let i=0;i<n;i++){
            if(s[i]=='(' || s[i]=='[' || s[i]=='{'){
                stack.push(s[i]);
            }
            else{
                let top=stack[stack.length-1];
                if(top=='(' && s[i]==')' || top=='[' && s[i]==']' || top=='{' && s[i]=='}'){
                    stack.pop();
                }
                else stack.push(s[i]);
            }
        }
        return stack.length==0;
    }
}

File: Data Structures & Algorithms/binary-tree-preorder-traversal/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> preorderTraversal(TreeNode root) {
     List<Integer>ansList = new ArrayList<>();
     if(root == null) return ansList;
     Stack<TreeNode>stack = new Stack<>();
     stack.add(root);
     while(!stack.isEmpty()){
        TreeNode top = stack.pop();
        ansList.add(top.val);
        if(top.right!=null) stack.add(top.right);
        if(top.left!=null) stack.add(top.left);
     }
     return ansList;   
    }
}